physics.py

```diff
@@ -0,0 +1,100 @@
+import math
+
+from pyfrc.physics.core import PhysicsInterface
+
+import ctre
+import navx
+from utilities.ctre import FALCON_CPR, FALCON_FREE_RPS
+
+from wpimath.kinematics import SwerveDrive4Kinematics
+
+from components.chassis import Chassis, SwerveModule
+from components.turret import Turret
+
+
+class PhysicsEngine:
+    def __init__(self, physics_controller: PhysicsInterface, robot: "MyRobot"):
+        self.physics_controller = physics_controller
+
+        # Swerve modules
+        self.chassis_1_drive = ctre.WPI_TalonFX(1)
```

Comment:
I think creating duplicate Talon instances is undefined behaviour. You'll want to access these from the `robot` that's passed in.

Reply:
When I did that it did not work properly - the motor movements were all borked.

I'll take another look.

Reply:
Weird, it worked when I changed it this time...